### PR TITLE
Sass 4211

### PR DESCRIPTION
**Pull request by georgievh**
### Description
Please include a brief summary of the change you've made and the issue that it fixes.

Add a link to the relevant story in Jira
[SASS-4211](https://jira.tools.tax.service.gov.uk/browse/SASS-4211)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/services/EmploymentSessionService.scala [hunk 1]
```diff
@@ -413,7 +413,7 @@ class EmploymentSessionService @Inject()(employmentUserDataRepository: Employmen
       priorDataResponse <- getPriorData(request.user, taxYear)
     } yield {
       if (optionalCya.isRight) {
```
Comment (nickgravgaard) [inline app/services/EmploymentSessionService.scala:415]: How about this instead:
```if (optionalCya == Right(None)) logger.info(s"[EmploymentSessionService][getAndHandle] No employment CYA data found for user. SessionId: ${request.user.sessionId}")```

File: app/services/EmploymentSessionService.scala [hunk 2]
```diff
@@ -439,7 +439,7 @@ class EmploymentSessionService @Inject()(employmentUserDataRepository: Employmen
...
```
Comment (nickgravgaard) [inline app/services/EmploymentSessionService.scala:441]: See above

File: app/services/EmploymentSessionService.scala [hunk 3]
```diff
@@ -521,7 +521,7 @@ class EmploymentSessionService @Inject()(employmentUserDataRepository: Employmen
       priorDataResponse <- getPriorData(request.user, taxYear)
     } yield {
       if (optExpensesUserData.isRight) {
```
Comment (nickgravgaard) [inline app/services/EmploymentSessionService.scala:523]: See above

File: app/services/EmploymentSessionService.scala (hunk 1 above)
Comment (georgievh) [reply app/services/EmploymentSessionService.scala:415]: There are many problems with this module. I did not introduce this code. I fixed many warnings, but I cant fix it all. :)

File: app/services/EmploymentSessionService.scala (hunk 2 above)
Comment (georgievh) [reply app/services/EmploymentSessionService.scala:441]: Same as above

File: app/services/EmploymentSessionService.scala (hunk 3 above)
Comment (georgievh) [reply app/services/EmploymentSessionService.scala:523]: Same as above

**Review by rtna-hmrc — APPROVED** [review]

**Pull request merged**